app/models/chargeable_field.rb

```diff
@@ -136,8 +138,10 @@ def self.seed_data
   end
 
   def self.chargeable_cols_on_metric_rollup
-    existing_cols = MetricRollup.attribute_names
-    chargeable_cols = pluck(:metric) & existing_cols
-    chargeable_cols.map! { |x| VIRTUAL_COL_USES[x] || x }.sort
+    @chargeable_cols_on_metric_rollup ||= begin
+      existing_cols = MetricRollup.attribute_names
+      chargeable_cols = pluck(:metric) & existing_cols
```

Comment:
@libor, I think this line may be causing the sporadic test failures. If this code is called before `ChargeableField` is seeded it will memoize an empty array. The other tests will use the memoized value and potentially get the nil error. I found that `consumption_with_rollups_spec.rb` calls this code in a bunch of places without seeding `ChargeableField`. The fix may be to just make sure the class is seeded first.

Reply:
yes! and finally I probably found a culprit. `./spec/models/chargeback/consumption_with_rollups_spec.rb` but I am not sure it if it is the only culprit.

thanks